src/serde/read_cache_lookup.rs

```diff
@@ -156,10 +156,13 @@ impl ReadCacheLookup {
                 if *node == self.root_hash {
                     // make sure we never return a path that needs more (or the
                     // same) bytes to serialize than the node we're referencing.
-                    if let Some(path_len) = atom_length_bits(path.len() as u64) {
-                        if path_len < max_bytes_for_path_encoding {
-                            let p = reversed_path_to_vec_u8(path);
-                            possible_responses.push(p);
+                    // path.len() + 1 is because reversed_path_to_vec_u8() will
+                    // also add the "terminator" bit, at the far left (MSB)
+                    // if we have 8 steps to traverse, we need 9 bits to represent
+                    // it as a path
+                    if let Some(path_len) = atom_length_bits(path.len() as u64 + 1) {
```

Comment:
this `+ 1` fixes a mistake. `reversed_path_to_vec_u8()` adds the left-most "terminator" bit to the path. That bit has to be taken into account when predicting how many bytes the path will be encoded as

Reply:
I wonder if the `+1` should be absorbed into the implementation of `.len`? What's the type of `path` here?

Reply:
it used to be `Vec<bool>` but I recently changed it to `BitVec`